biz.aQute.bndlib/src/aQute/bnd/cdi/AnnotationReader.java

```diff
@@ -0,0 +1,340 @@
+				&& !clazz.is(QUERY.ANNOTATED, DEPENDENT_INSTR, analyzer)
+				&& !clazz.is(QUERY.ANNOTATED, INTERCEPTOR_INSTR, analyzer)
+				|| clazz.is(QUERY.IMPLEMENTS, EXTENSION_INSTR, analyzer)) {
+
+				return null;
+			}
+
+			// check for @Vitoed package
+			Clazz packageClazz = analyzer.getPackageInfo(clazz.getClassName()
+				.getPackageRef());
+			if (packageClazz != null && packageClazz.is(QUERY.ANNOTATED, VETOED_INSTR, analyzer)) {
+				return null;
+			}
+		}
+
+		clazz.parseClassFileWithCollector(this);
+
+		// the default discovery mode is 'annotated_by_bean' to indicate that
+		// classes annotated with @Bean or in packages annotated with @Beans are
+		// considered
+		if (options.contains(Options.annotated_by_bean) && !definitions.get(0).marked) {
+			return null;
+		}
+
+		return definitions;
+	}
+
+	class PackageDef extends ClassDataCollector {
+		Instruction marked;
+
+		@Override
+		public void annotation(Annotation annotation) throws Exception {
+			java.lang.annotation.Annotation a = annotation.getAnnotation();
+
+			if (a instanceof Beans) {
+				Object[] beanClasses = annotation.get("value");
+
+				if (beanClasses != null && beanClasses.length > 0) {
+					marked = new Instruction(Arrays.stream(beanClasses)
+						.map(c -> c.toString())
+						.collect(Collectors.joining(",")));
+				} else {
+					marked = new Instruction("*");
+				}
+			}
+		}
+	}
+
+	@Override
+	public void annotation(Annotation annotation) {
+		try {
+			java.lang.annotation.Annotation a = annotation.getAnnotation();
+
+			if (a instanceof Bean) {
+				definitions.get(0).marked = true;
+			}
+			else if (a instanceof Service) {
+				doService(annotation, a);
+			}
+			else if (a instanceof MinimumCardinality) {
+				minimumCardinality = (int) annotation.get("value");
+				if (minimumCardinality > 0 && referenceDef != null) {
+					referenceDef.cardinality = ReferenceCardinality.AT_LEAST_ONE;
+				}
+			}
+			if (a instanceof Reference) {
+				doReference(annotation, a);
+			}
+		} catch (Exception e) {
+			analyzer.exception(e, "During bean processing on class %s, exception %s", clazz, e);
+		}
+	}
+
+	private void doReference(Annotation annotation, java.lang.annotation.Annotation a) {
+		referenceDef = new ReferenceDef(annotation, member, analyzer);
+		if (minimumCardinality > 0) {
+			referenceDef.cardinality = ReferenceCardinality.AT_LEAST_ONE;
+			minimumCardinality = -1;
+		}
+		definitions.get(0).references.add(referenceDef);
+	}
+
+	private void doService(Annotation annotation, java.lang.annotation.Annotation a) {
+		switch (annotation.getElementType()) {
+			case FIELD :
+				if (member instanceof Clazz.FieldDef) {
+					String signature = member.getSignature();
+					if (signature == null) {
+						signature = member.getDescriptor()
+							.toString();
+					}
+					FieldSignature sig = FieldSignature.of(signature);
+					ClassTypeSignature returnType = (ClassTypeSignature) sig.type;
```

Comment:
Use a FieldResolver to get the field type. See
https://github.com/bndtools/bnd/blob/ceba0425550d6bd6bfb572e5419488cc042ddfd9/biz.aQute.bndlib/src/aQute/bnd/component/AnnotationReader.java#L958-L960

You must also test the result is an instanceof ClassTypeSignature. It may not be and your case will fail.